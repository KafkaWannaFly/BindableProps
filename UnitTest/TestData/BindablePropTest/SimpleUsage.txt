namespace UnitTest.TestData.BindablePropTest
{
    public partial class SimpleUsage
    {
        [BindableProp]
        string title;

        [BindableProp]
        string description = "To the moon";
    }
}
