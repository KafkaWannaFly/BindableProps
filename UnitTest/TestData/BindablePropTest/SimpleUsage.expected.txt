// <auto-generated/>

namespace UnitTest.TestData.BindablePropTest
{
    public partial class SimpleUsage
    {
        public static readonly BindableProperty TitleProperty = BindableProperty.Create(
            nameof(Title),
            typeof(string),
            typeof(SimpleUsage),
            default,
            (BindingMode)0,
            null,
            (bindable, oldValue, newValue) => 
                        ((SimpleUsage)bindable).Title = (string)newValue,
            null,
            null,
            null
        );

        public string Title
        {
            get => title;
            set 
            {
                OnPropertyChanging(nameof(Title));

                title = value;
                SetValue(SimpleUsage.TitleProperty, title);

                OnPropertyChanged(nameof(Title));
            }
        }

        public static readonly BindableProperty DescriptionProperty = BindableProperty.Create(
            nameof(Description),
            typeof(string),
            typeof(SimpleUsage),
            "To the moon",
            (BindingMode)0,
            null,
            (bindable, oldValue, newValue) => 
                        ((SimpleUsage)bindable).Description = (string)newValue,
            null,
            null,
            null
        );

        public string Description
        {
            get => description;
            set 
            {
                OnPropertyChanging(nameof(Description));

                description = value;
                SetValue(SimpleUsage.DescriptionProperty, description);

                OnPropertyChanged(nameof(Description));
            }
        }
    }
}
